### PR TITLE
improve CA and certificate generation

### PR DESCRIPTION
**Pull request by joelanford**
**Motivation for the change:**
Recently during an audit on a user's cluster, it was discovered that
OLM's certificate generation functionality has a few minor shortcomings.
  1) The generated CA and server cert do not include a common name,
     which causes some tooling to have trouble tracing the cert chain.
  2) The generated CA and server cert include unnecessary key usages,
     which means those certificates can be used for more than their
     intended purposes.

**Description of the change:**
This commit resolves the above issues by ensuring the certificates
include common names and by using the minimal key usages necessary.

**Testing remarks:**
Thus far, I have only manually tested via `run-local` and verified that:
1) The generated packageserver secret contained the expected cerificates.
2) The packagemanifests API worked (thus proving the kube-apiserver can use OLM's generated CA to verify OLM's generated cert for the packageserver api service.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

### CODE REVIEW

**Review by dinhxuanvu — APPROVED** [review]
/approve

File: pkg/controller/certs/certs.go [hunk 1]
```diff
@@ -120,12 +120,12 @@ func CreateSignedServingPair(notAfter time.Time, organization string, ca *KeyPai
 	certDetails := &x509.Certificate{
 		SerialNumber: serial,
 		Subject: pkix.Name{
+			CommonName:   hosts[0],
```
Comment (timflannagan) [inline pkg/controller/certs/certs.go:123]: Do we do any length checking for the hosts array length anywhere?

**Comment by timflannagan** [comment]
Just one quick question, but otherwise this lgtm.

File: pkg/controller/certs/certs.go (hunk 1 above)
Comment (joelanford) [reply pkg/controller/certs/certs.go:123]: Right now, it always has exactly two items:

0. `<svcName>.<svcNamespace>`
1. `<svcName>.<svcNamespace>.svc`

Comment (timflannagan) [reply pkg/controller/certs/certs.go:123]: Alright, works for me.

/lgtm

**Pull request merged**